pkg/errors/sentry.go

```diff
@@ -32,11 +32,11 @@ func CaptureDiagnostics(dd diag.Diagnostics, tags map[string]string) {
 			continue
 		}
 		sentry.WithScope(func(scope *sentry.Scope) {
-			if t, ok := d.(core.FetchDiagnostic); ok {
-				if allowUnmanaged && t.Version == plugin.Unmanaged {
+			if ok, p, v := isFetchDiagnostic(d); ok {
+				if !allowUnmanaged && v == plugin.Unmanaged {
 					return
 				}
-				scope.SetTags(map[string]string{"provider": t.Provider, "provider_version": t.Version, "resource": t.Description().Resource})
+				scope.SetTags(map[string]string{"provider": p, "provider_version": v, "resource": d.Description().Resource})
```

Comment:
Thought about making a `TaggedDiagnostic` to embed tags but turns out we also use `IsConfigureDiag` to determine some things about error classification.

Reply:
Done in [1c11f31](https://github.com/cloudquery/cloudquery/pull/744/commits/1c11f3130a446a99761ee4b072da8239fd288b18)